tests/simple/testdata/proto2.textproto

```diff
@@ -528,3 +528,148 @@ section {
+    value { bool_value: false }
+  }
+  test {
+    name: "map_one_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{map_string_string: {'MT': ''}}.map_string_string)"
+    value { bool_value: true }
+  }
+  test {
+    name: "map_one"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{map_string_string: {'one': 'uno'}}.map_string_string)"
+    value { bool_value: true }
+  }
+  test {
+    name: "map_many"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{map_string_string: {'one': 'uno', 'two': 'dos'}}.map_string_string)"
+    value { bool_value: true }
+  }
+  test {
+    name: "required"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestRequired{required_int32: 4}.required_int32)"
+    value { bool_value: true }
+  }
+  test {
+    name: "optional_unset_no_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{}.single_sint32)"
+    value { bool_value: false }
+  }
+  test {
+    name: "optional_set_no_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{single_sint32: -4}.single_sint32)"
+    value { bool_value: true }
+  }
+  test {
+    name: "optional_unset_with_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{}.single_int32)"
+    value { bool_value: false }
+  }
+  test {
+    name: "optional_set_with_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{single_int32: 16}.single_int32)"
+    value { bool_value: true }
+  }
+  test {
+    name: "optional_set_to_default"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{single_int32: -32}.single_int32)"
+    value { bool_value: true }
+  }
+  test {
+    name: "optional_message_unset"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{}.standalone_message)"
+    value { bool_value: false }
+  }
+  test {
+    name: "optional_message_set"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{standalone_message: TestAllTypes.NestedMessage{}}.standalone_message)"
+    value { bool_value: true }
+  }
+  test {
+    name: "optional_enum_unset"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{}.standalone_enum)"
+    value { bool_value: false }
+  }
+  test {
+    name: "optional_enum_set"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{standalone_enum: TestAllTypes.NestedEnum.BAR}.standalone_enum)"
+    value { bool_value: true }
+  }
+  test {
+    name: "oneof_unset"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{}.single_nested_message)"
+    value { bool_value: false }
+  }
+  test {
+    name: "oneof_other_set"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "has(TestAllTypes{single_nested_enum: TestAllTypes.NestedEnum.BAZ}.single_nested_message)"
```

Comment:
Could you add a test for when an enum at ordinal 0 is set in proto2 and proto3? NestedEnum.FOO, I think.

Reply:
Done, and might have found more issues with cel-go.